pull-submodules.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+set -o nounset -o errexit
```

Comment:
Upon second thought, I think this script will be tempting to a lot of students, when the submodule-based route is really what they want. Maybe there should be a short explainer and an "are you sure that this is what you want?" prompt.

Reply:
Good idea. I added a warning and force the user to run this script with the `-x` option if they are sure they want to execute it.